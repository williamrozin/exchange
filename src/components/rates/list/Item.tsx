import { ListItem } from '@material-ui/core'
import React, { ReactNode, SFC } from 'react'
import styled from 'styled-components'

interface IProps {
    index: number
    children: ReactNode
}

interface IWrapper {
    index: IProps['index']
}

const Wrapper = styled.div<IWrapper>`
    cursor: pointer;
    && {
        .list-item {
            background-color: ${(props) => props.index % 2 === 0
                ? '#EEEEEE'
                : '#F5F5F5'
            };
        }

        &:hover {
            .list-item {
                background-color: #E0E0E0;
            }
        }
    }
`

const Item: SFC<IProps> = (props) =>
    <Wrapper index={ props.index }>
        <ListItem className='list-item'>
            { props.children }
        </ListItem>
    </Wrapper>

export default Item
