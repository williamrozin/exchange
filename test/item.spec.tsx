import { shallow } from 'enzyme'
import React from 'react'

import Item from '../src/components/rates/list/Item'

describe('Item component', () => {
    it('should render an item with an even index', () => {
        const item = shallow(<Item index={ 0 }>Even item</Item>)
        expect(item.contains('Even item')).toBeTruthy()
        expect(item.contains('"backgroundColor": "#EEEEEE"'))
        expect(item).toMatchSnapshot()
    })
    it('Should render an Item with an odd index', () => {
        const item = shallow(<Item index={ 1 }>Odd item</Item>)
        expect(item.contains('Odd item')).toBeTruthy()
        expect(item.contains('"backgroundColor": "#F5F5F5"'))
        expect(item).toMatchSnapshot()
    })
})
