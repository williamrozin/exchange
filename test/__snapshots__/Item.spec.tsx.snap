// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component Should render an Item with an odd index 1`] = `
<WithStyles(ListItem)
  style={
    Object {
      "backgroundColor": "#F5F5F5",
    }
  }
>
  Odd item
</WithStyles(ListItem)>
`;

exports[`Item component should render an item with an even index 1`] = `
<WithStyles(ListItem)
  style={
    Object {
      "backgroundColor": "#EEEEEE",
    }
  }
>
  Even item
</WithStyles(ListItem)>
`;
